main.js

```diff
@@ -1,62 +1,59 @@
 class Game {
     constructor() {
         this.imageOnClickBinded = this.imageOnClick.bind(this);
-        this.imagesRandomised = [];
-        this.nowOpened = 0;      
         this.timer=new Timer();
-        this.DOMManagerInst = new DOM_Manager();
-        this.DOMManagerInst.setListeners(this.imageOnClickBinded);
-        this.numCards = this.DOMManagerInst.cardsCount;
-        this.fillImagesRandom();  
     }
-
-    fillImagesRandom(count) {
+    start(cards) {
+        this.cards = cards;
+        this.cardsCount=cards.length;
+        this.imagesRandomised = [];
+        this.nowOpened = 0;
+        this.fillImagesRandom();
+        this.timer.startTimer();
+    }
+    fillImagesRandom() {
         const imageNumbers = [];
-        for(let i=0; i<6;i++) {
+        for(let i=0; i<this.cardsCount/2;i++) {
             imageNumbers.push(i);
             imageNumbers.push(i);
         }
-        for(let i=this.numCards-1;i>=0;i--) {
+        for(let i=this.cardsCount-1;i>=0;i--) {
             let num = Math.round(Math.random() * i);
             this.imagesRandomised.push(imageNumbers[num]);
             imageNumbers.splice(num, 1);
         }
+        let i=0;
+        for(let card of this.cards) {
+            card.fileName=this.imagesRandomised[i];
+            i++;
+        }
     }
+ 
     imageOnClick(idClicked) {
-        if(!this.timer.isStarted) {
-            this.timer.startTimer();
-        }
         if(idClicked === this.prevClicked) return;
-        this.nowOpened++;
-        if(this.nowOpened > 2) {
-            this.nowOpened = 2;
-            return;
-        }
-        this.DOMManagerInst.showFace(idClicked, this.imagesRandomised[idClicked]);
-        if(this.nowOpened===1) {
+        if(this.cards.filter(a=>a.opened===true).length === 2) return;

```

Comment:
Use _Variable  extraction_ or/and _Method Extraction_ (getter) for 
```js
this.cards.filter(a=>a.opened===true).length
```
https://refactoring.guru/uk/extract-variable
https://refactoring.guru/uk/extract-method